pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1158,7 +1158,16 @@ func (m *ManagerImpl) ShouldResetExtendedResourceCapacity() bool {
 	if err != nil {
 		return false
 	}
-	return len(checkpoints) == 0
+
+	// Currently, the checkpoint file name for the device plugin is only kubelet_internal_checkpoint.
+	// If there are any changes in the future, corresponding adjustments will be needed.
+	for _, checkpointName := range checkpoints {
+		if checkpointName == kubeletDeviceManagerCheckpoint {
+			return false
+		}
+	}
```

Comment:
Would it be better to fix `ListCheckpoints` instead, so it would return a proper list of checkpoints?

Reply:
I feel like there isn't a particularly reasonable place to put it. The main reason is that checkpointdir shares the directory with other files (such as kubelet.sock、device plugin sock), and currently, there is only one checkpoint file (`kubelet_internal_checkpoint`), `ListCheckpoints` always returns only kubelet_internal_checkpoint.

Reply:
> Would it be better to fix `ListCheckpoints` instead, so it would return a proper list of checkpoints?

1. I will adopt this suggestion for now. 
1. I will try to fix the issue using `GetCheckpoint` in `ShouldResetExtendedResourceCapacity`

Reply:
`ListCheckpoints` is a reasonable place in my opinion. It should filter out sockets as sockets are not checkpoints.

Reply:
@zhifei92 please address above review comment, thanks!

Reply:
> It should filter out sockets as sockets are not checkpoints.

I fully agree with this approach, thank you.